icu4c/source/i18n/measunit_extra.cpp

```diff
@@ -765,7 +765,9 @@ bool appendImpl(MeasureUnitImpl& impl, const SingleUnitImpl& unit, UErrorCode& s
 void appendAndMergeImpl(MeasureUnitImpl &impl, const SingleUnitImpl &unit, UErrorCode &status) {
     for (int32_t i = 0, n = impl.units.length(); i < n; i++) {
         auto *candidate = impl.units[i];
-        if (candidate->identifier == unit.identifier && candidate->siPrefix == unit.siPrefix) {
```

Comment:
I'm not actually sure if this used to be a pointer comparison instead of a string-contents comparison - I think so though. (If it was a StringPiece, it might have been a more sophisticated "operator==" at work).

For the new code, I'm first doing a pointer comparison, since by the current implementation they should in fact match, and if not matching, I'm comparing the string contents (for robustness against implementation changes).

Double-calling getSimpleUnitID though - the compiler may not be smart enough to factor that out. Should I assign these to variables, maybe?

Reply:
FYI, the fastest and safest thing to use to compare the simple unit IDs of two SimpleUnitImpls is to compare the `.index` property.

Reply:
you can use [`strcmp` function](http://www.cplusplus.com/reference/cstring/strcmp/)

Reply:
Dropped the strcmp function, going with comparing .index - seems I didn't see the forest for the trees. ;-) Thanks for catching the oversight!